Rst help cheatsheet

SenorGeese opened the pull request:
I was starting to forget rst syntax and tricks so I made this, figured if i'm going to make this. Might as well make it public.

laundmo commented:
This is great, but i would maybe prefer it in a seperate document. if you could move that over, i'll merge.

SenorGeese commented:
Alright, will do.

**My TODO (because i forgor too much):**
- Clean up the RST TESTING DANGER
- Move to separate page
- Fix Header 1 appearing on ToC
- Add Links, Hyperlinks, Relative Links
- Add References (the thing to make table **much** better

SenorGeese commented:
Because I'm procrastinating from doing my C++ classwork. I did a commit today smh.
Anyways,
- Cleaned up Danger Zone.
- Moved to separate page
- Fixed header problem

TODO
- Links
- References
- Tables
- maybe Directives (just to make reading other docs easier)

laundmo commented:
Having done a proper read through this, i really like how many things are in here. theres 2 concerns i have:

- a lot of the more complex features don't have a codeblock showing their usage, it would be neat to have code blocks for the full usage of some more elements.
- i don't thing the conversational style fits with most of the docs - i would much prefer a more direct style. "I prefer" etc can be replaced with "a recommended style" or similar.

i would prefer to get this merged soon, since I want to start work on #100 which will move around almost all files

SenorGeese commented:
- Which features don't have code blocks? Only feature missing a code block (or plain text) are the tables. Which are unfinished/not done.
- Hmm. I always thought "I prefer" was weak and says without confidence, if you prefer it though. Sure, I'll make those changes.
Okay. I'll try getting this to a mergeable state.

SenorGeese commented:
Found these useful, saving them here.
toctree little hints
- https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#include-other-rst-files-with-the-toctree-directive
- https://devguide.python.org/documentation/markup/

GeckoEidechse commented on docs/source/guides/contributing.rst:
```diff
@@ -1,16 +1,12 @@
-===========================
-
-All contributions to ModdingDocs are welcome. To add a change simply make a pull request to the `ModdingDocs repo <https://github.com/R2Northstar/ModdingDocs/>`_.
+=====
```
What's with the changes in `contributing.rst`? They seems to be mostly formatting related. Should be separate PR IMO.

SenorGeese replied:
Since the rst cheatsheet was originally following the contributing section. All the previous header organizations was incorrect and broken. So I fixed it. Then I was told to make it into another file.

GeckoEidechse replied:
Uh can we maybe turn that into a separate PR then and keep this file here as is in this PR?

Separating things out helps with reviewing and makes reverting changes easier if need be ^^

laundmo commented:
i'm mostly fine with merging this soon, but who owns the copyright to the cat picture?

SenorGeese commented:
> c

Not sure. I just googled and found it. I could replace it a paint.exe drawing ig? Unless you have a better recommendation.

SenorGeese commented:
Replaced the cat image with my own drawing.

GeckoEidechse commented on docs/source/guides/rsthelp.rst:
```diff
@@ -0,0 +1,476 @@
+  hi     elo      |title|
+====== ====== ============================
+
+Complex Complex:
+
+TODO
+
+Complex:
+
+TODO
+
+Complex Easiest:
+
+TODO
```
More `TODO`s ^^

SenorGeese replied:
Just noting down the organization for future PR or commits.

SenorGeese commented on docs/source/guides/rsthelp.rst:
```diff
@@ -0,0 +1,476 @@
+
+.. image:: /_static/rsthelp/mpt.png
+    :height: 40
+.. danger:: **DANGER ZONE BELOW**
+.. danger:: **DANGER ZONE BELOW**
+.. danger:: **DANGER ZONE BELOW**
+.. danger:: **DANGER ZONE BELOW**
+.. danger:: **DANGER ZONE BELOW**
```
Everything below that point is not properly documented and just functional .rst things.

SenorGeese commented on docs/source/guides/rsthelp.rst:
```diff
@@ -0,0 +1,476 @@
+References:
+-----
+
+TODO - REALLY great for tables. Basically text links (i haven't done much testing).
```
It's not finished.

SenorGeese commented on docs/source/guides/rsthelp.rst:
```diff
@@ -0,0 +1,476 @@
+Examples:
+#####
+
+.. attention:: This is done ``.. attention:: YOUR_TEXT HERE``
```
It could be, I think "This is done" is still understandable being I didn't want to write to many words.

SenorGeese commented on docs/source/guides/rsthelp.rst:
```diff
@@ -0,0 +1,476 @@
+.. important:: Resize images and images within figures with ``:width:``, ``:height:``, or ``:scale:``.
+
+Figures are kind of like their own section/area on the page. 
+Blah Blah Blah, we can do more with figures, basically whatever you can do with images but a little more.
```
It's like an expression ig? Just like saying "I don't need to explain why having more functionality is better or how and why" I just need you to know that figures are better.

SenorGeese commented on docs/source/guides/rsthelp.rst:
```diff
@@ -0,0 +1,476 @@
+Links:
+-----
+
+TODO - Single use, and Indexed links.
```
Yes it's not finished.

GeckoEidechse commented:
bump

GeckoEidechse commented:
With the move to markdown based docs, this is no longer necessary.
